tango/step_graph.py

```diff
@@ -151,28 +185,77 @@ def __iter__(self) -> Iterator[str]:
         """
         return iter(self.parsed_steps)
 
-    def ordered_steps(self) -> List[Step]:
+    @classmethod
+    def ordered_steps(cls, step_dict: Dict[str, Step]) -> List[Step]:
         """
         Returns the steps in this step graph in an order that can be executed one at a time.
 
         This does not take into account which steps may be cached. It simply returns an executable
         order of steps.
         """
+        dependencies = {
+            step_name: set([dep.name for dep in step.dependencies])
+            for step_name, step in step_dict.items()
+        }
         result: List[Step] = []
-        steps_run: Set[Step] = set()
-        steps_not_run = list(self.parsed_steps.values())
-        while len(steps_not_run) > 0:
-            step = steps_not_run.pop(0)
-            if len(step.dependencies & steps_run) == len(step.dependencies):
-                steps_run.add(step)
-                result.append(step)
-            else:
-                steps_not_run.append(step)
+        for step_name in cls._get_ordered_steps(dependencies):
+            step_dict[step_name].name = step_name
+            result.append(step_dict[step_name])
         return result
 
+    def find_uncacheable_leaf_steps(self) -> Set[Step]:
```

Comment:
Let's be consistent about this kind of naming. It's not `Step.get_workspace()`, it's just `Step.workspace()`. So it should just be `uncacheable_leaf_steps()` here, like it already is for `ordered_steps()` above. Same for `sub_graph()`.